mlir/utils/performance/ck-benchmark-driver/ck-atten-benchmark-driver.cpp

```diff
@@ -0,0 +1,505 @@
+    const auto oBytes = benchmark::getByteSize(options.elementType, oElems);
+    const auto biasBytes =
+        benchmark::getByteSize(options.elementType, biasElems);
+
+    qHost = benchmark::allocAndFill(options.elementType, qBytes);
+    kHost = benchmark::allocAndFill(options.elementType, kBytes);
+    vHost = benchmark::allocAndFill(options.elementType, vBytes);
+    oHost = benchmark::allocAndFill(options.elementType, oBytes);
+    biasHost = benchmark::allocAndFill(options.elementType, biasBytes);
+
+    qDevice = benchmark::getGpuBuffer(qHost, qBytes);
+    kDevice = benchmark::getGpuBuffer(kHost, kBytes);
+    vDevice = benchmark::getGpuBuffer(vHost, vBytes);
+    oDevice = benchmark::getGpuBuffer(oHost, oBytes);
+    biasDevice = benchmark::getGpuBuffer(biasHost, biasBytes);
+  }
+  ~Data() {
+    free(qHost);
+    free(kHost);
+    free(vHost);
+    free(oHost);
+    free(biasHost);
+
+    HIP_ABORT_IF_FAIL(hipFree(qDevice));
+    HIP_ABORT_IF_FAIL(hipFree(kDevice));
+    HIP_ABORT_IF_FAIL(hipFree(vDevice));
+    HIP_ABORT_IF_FAIL(hipFree(oDevice));
+    HIP_ABORT_IF_FAIL(hipFree(biasDevice));
+  }
+
+  void *qHost{nullptr};
+  void *kHost{nullptr};
+  void *vHost{nullptr};
+  void *oHost{nullptr};
+  void *biasHost{nullptr};
+
+  void *qDevice{nullptr};
+  void *kDevice{nullptr};
+  void *vDevice{nullptr};
+  void *oDevice{nullptr};
+  void *biasDevice{nullptr};
+
+  std::vector<ck::index_t> qLengths{};
+  std::vector<ck::index_t> kLengths{};
+  std::vector<ck::index_t> vLengths{};
+  std::vector<ck::index_t> oLengths{};
+  std::vector<ck::index_t> biasLengths{};
+
+  std::vector<ck::index_t> qStrides{};
+  std::vector<ck::index_t> kStrides{};
+  std::vector<ck::index_t> vStrides{};
+  std::vector<ck::index_t> oStrides{};
+  std::vector<ck::index_t> biasStrides{};
+
+private:
+  Options options;
+};
+
+template <typename T>
+std::ostream &operator<<(std::ostream &os, std::vector<T> &v) {
+  for (auto &item : v) {
+    os << item << ", ";
+  }
+  return os;
+}
+
+std::ostream &operator<<(std::ostream &os, Data &data) {
+  os << "Q\n\t"
+     << "lengths: " << data.qLengths << "\n\tstrides: " << data.qStrides
+     << "\n";
+  os << "K\n\t"
+     << "lengths: " << data.kLengths << "\n\tstrides: " << data.kStrides
+     << "\n";
+  os << "V\n\t"
+     << "lengths: " << data.vLengths << "\n\tstrides: " << data.vStrides;
+  return os;
+}
+
+// Note, this implementation is in sync with the one from
+// `perfRunner.py` (see `computeTFlops`)
+double computeBaseFlopCount(const Options &options) {
+  const auto g{static_cast<double>(options.groupSize)};
+  const auto SQ{static_cast<double>(options.sequenceLengthQ)};
+  const auto SK{static_cast<double>(options.sequenceLengthK)};
+  const auto HQK{static_cast<double>(options.headDimsQK)};
+  const auto HV{static_cast<double>(options.headDimsV)};
+
+  const auto gemm1 = 2.0 * SQ * HQK * SK;
+  const auto scale = options.withAttnScale ? SQ * SK : 0.0;
```

Comment:
do we need a something similiar for bias?

Reply:
It is handled in `struct AttentionWithBias`. It is here

https://github.com/ROCm/rocMLIR/blob/28759faa2827502e8725caeb14b4d80089ba2a97/mlir/utils/performance/ck-benchmark-driver/ck-atten-benchmark-driver.cpp#L355-L364

Reply:
ah ok. missed that.